go/runtime/txpool/txpool.go

```diff
@@ -460,20 +466,49 @@ func (t *txPool) checkTxBatch(ctx context.Context, rr host.RichRuntime) {
 		return
 	}
 
+	// Check if node is synced.
+	var notReady bool
+	select {
+	case <-t.consensus.Synced():
+		// Node is synced.
+	default:
+		// Node is not synced.
+		notReady = true
+		t.logger.Debug("consensus is not synced yet")
+	}
+
+	if !notReady {
+		// Node is synced, ensure block round is synced to storage.
+		waitSyncCtx, cancelWaitSyncCtx := context.WithTimeout(ctx, checkTxWaitRoundSyncedTimeout)
+		defer cancelWaitSyncCtx()
+
+		t.logger.Debug("ensuring block round is synced", "round", bi.RuntimeBlock.Header.Round)
+		if _, err = t.history.WaitRoundSynced(waitSyncCtx, bi.RuntimeBlock.Header.Round); err != nil {
+			notReady = true
+			t.logger.Debug("block round is not synced yet", "err", err)
+		}
+	}
+
+	if notReady {
+		// Node or block round aren't synced yet, so make sure the batch
```

Comment:
I would say that the  `checkWorker` should wait for consensus sync, as this is a necessary precondition for all checks. So you don't need an extra timeout and the pool can start processing immediately after sync.

Reply:
It still needs to wait for block sync, though.

Reply:
Yes, wait here for block sync, and for consensus there.

Reply:
But if the node loses consensus sync, we still need to check for it here, so we don't blame the runtime.  I can add an additional wait for consensus sync into `checkWorker`, but we still need to retain it here, I think?

Reply:
Loses sync? I think that once consensus is synced, the call to `Consensus.Synced()` will always return a closed channel.

Reply:
Oh, OK, I will make it wait in the worker and just check for block sync here then.

Reply:
Done, please have another look :)